proposals/NNNN-clock-instant-date-duration.md

```diff
@@ -0,0 +1,498 @@
+
+The storage change is not a hard requirement; and may be a point in which we might decide is not worth taking.
+
+All remaining APIs on Date will exist still at the Foundation layer for compatibility with existing software.
+
+To be clear; we are not suggesting that Calendar, Locale, or TimeZone be moved down; those transitions are distinctly out of scope of this proposal and are not a goal.
+
+##### WallClock
+
+Wall clocks are useful since they represent a transmittable form of time. Instants can be serialized and sent from one machine to another and the values are meaningful in a foreign context. That transmission can be immediately useful when dealing with concepts like distributed actors; where an actor may be hosted on a remote machine and a deadline for work is sent across from one domain to another. The `WallClock` type will use `Date` as its `Instant` type and provide an extension to access the clock instance as the inferred base type property.
+
+```swift
+public struct WallClock {
+  public init()
+  
+  public static var now: Date { get }
+}
+
+extension WallClock: ClockProtocol {
+  public typealias Instant = Date
+
+  public var now: Date { get }
+  public func sleep(until deadline: Date) async throws
+}
+
+extension ClockProtocol where Self == WallClock {
+  public static var wall: WallClock { get }
+}
+```
+
+##### MonotonicClock
+
+When instants are for local processing only and need to be high resolution without the encumbrance of suspension while the machine is asleep `MonotonicClock` is the tool for the job. The `MonotonicClock.Instant` type can be initialized with a wall clock instant if that value can be expressed in terms of a relative point to now; knowing the delta between the current time and the specified wall clock instant a conversion to the current monotonic reference point can be made such that conversion (if possible) represents what the value would be in terms of the monotonic clock. Much like the wall clock version the monotonic clock also offers an extension to access the clock instance as the inferred base type property.
+
+```swift
+public struct MonotonicClock {
+  public init()
+  
+  public static var now: Instant { get }
+}
+
+extension MonotonicClock: ClockProtocol {
+  public struct Instant { 
+    public init?(converting wallclockInstant: WallClock.Instant)
+    
+    public static var now: MonotonicClock.Instant { get }
```

Comment:
`InstantProtocol` doesn't have a `now` requirement; is this intentionally public?

Reply:
Yep, it is so you can use the .now shorthand for the common clock instant types